src/test/java/de/caritas/cob/userservice/api/service/emailsupplier/TenantDataSupplierTest.java

```diff
@@ -0,0 +1,64 @@
+package de.caritas.cob.userservice.api.service.emailsupplier;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.when;
+
+import de.caritas.cob.userservice.api.tenant.TenantContext;
+import de.caritas.cob.userservice.api.tenant.TenantData;
+import de.caritas.cob.userservice.mailservice.generated.web.model.TemplateDataDTO;
+import de.caritas.cob.userservice.tenantservice.generated.web.TenantControllerApi;
+import de.caritas.cob.userservice.tenantservice.generated.web.model.RestrictedTenantDTO;
+import java.util.List;
+import org.jeasy.random.EasyRandom;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.test.util.ReflectionTestUtils;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TenantDataSupplierTest {
+
+  @InjectMocks
+  TenantDataSupplier tenantDataSupplier;
+
+  @Mock
+  TenantControllerApi tenantControllerApi;
+
+  private final EasyRandom easyRandom = new EasyRandom();
+
+  @Test
+  public void test_Provide_Tenant_Specific_Data_For_Email_Templates() {
```

Comment:
please follow the convention for unit test method naming. 
And for this particular test I recommend to split sections with // given // when // then, for readability

Reply:
done